root/programs/C#/Frameworks/Infrastructure/Public/Util/CodeSigningXML.cs

```diff
@@ -0,0 +1,274 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. 
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#endregion
+
+//**********************************************************************************
+//* クラス名        ：CodeSigning
+//* クラス日本語名  ：CodeSigningクラス
+//*
+//* 作成者          ：生技 西野
+//* 更新履歴        ：
+//*
+//*  日時        更新者            内容
+//*  ----------  ----------------  -------------------------------------------------
+//*  2017/01/10  西野  大介        新規作成
+//**********************************************************************************
+
+// System
+using System;
+
+// 業務フレームワーク（循環参照になるため、参照しない）
+// フレームワーク（循環参照になるため、参照しない）
+
+// 部品
+using Touryo.Infrastructure.Public.Db;
+using Touryo.Infrastructure.Public.IO;
+using Touryo.Infrastructure.Public.Log;
+using Touryo.Infrastructure.Public.Str;
+using Touryo.Infrastructure.Public.Util;
+
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Touryo.Infrastructure.Public.Util
+{
+    /// <summary>
+    /// CodeSigningクラス
+    /// - RSACryptoServiceProvider:
+    ///   MD5, SHA1, SHA256, SHA384, SHA512
+    /// - DSACryptoServiceProvider:SHA1
+    /// だけ、サポート。
+    /// </summary>
+    public class CodeSigningXML : CodeSigning
+    {
+        // 署名の場合は、秘密鍵で署名して、公開鍵で検証。
+
+        #region mem & prop & constructor
+
+        /// <summary>AsymmetricAlgorithm</summary>
+        private AsymmetricAlgorithm _aa = null;
+        /// <summary>HashAlgorithm</summary>
+        private HashAlgorithm _ha = null;
+
+        /// <summary>
+        /// XMLPrivateKey
+        /// RFC 3275のXML秘密鍵
+        /// </summary>
+        public string XMLPrivateKey { get; set; }
+
+        /// <summary>
+        /// XMLPublicKey
+        /// RFC 3275のXML公開鍵
+        /// </summary>
+        public string XMLPublicKey { get; set; }
+
+        /// <summary>
+        /// Constructor
+        /// RFC 3275のXMLからキーペアを設定する。
+        /// </summary>
+        public CodeSigningXML(EnumCodeSigningAlgorithm eaa)
+        {
+            this.CreateAsymmetricAlgorithmServiceProvider(eaa, out this._aa, out this._ha);
+
+            if (string.IsNullOrEmpty(this.XMLPrivateKey))
+            {
+                // 秘密鍵をXML形式で取得
+                this.XMLPrivateKey = this._aa.ToXmlString(true);
+                // 公開鍵をXML形式で取得
+                this.XMLPublicKey = this._aa.ToXmlString(false);
+            }
+        }
+
+        #endregion
+
+        #region 署名(XML)
+
+        /// <summary>Sign</summary>
```

Comment:
@daisukenishino 

The term "Sign" is ambiguous.
(Does this method actually *sign*? Or, does this method create *digital signature*?)

At first sight, this method seems to create *digital signature*.
So, please write *'Create digital signature'* in *summary* tag and relating places.